From 437b6cdf50922dbbcb6de9a56e02af96aeb354c1 Mon Sep 17 00:00:00 2001
From: szszqkjyxgs_17014 <focalcrest.com>
Date: Fri, 20 Jun 2025 00:39:06 +0800
Subject: [PATCH] Revert "MALI: bifrost: Resolve a runtime WARNING when
 CONFIG_FORTIFY_SOURCE is enabled"

This reverts commit cf18504ebe450aa7492f390b3cfd416cf0b7b7db.
---
 drivers/gpu/arm/bifrost/csf/mali_kbase_csf_trace_buffer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/arm/bifrost/csf/mali_kbase_csf_trace_buffer.c b/drivers/gpu/arm/bifrost/csf/mali_kbase_csf_trace_buffer.c
index 707be020e314..46872f937dbf 100644
--- a/drivers/gpu/arm/bifrost/csf/mali_kbase_csf_trace_buffer.c
+++ b/drivers/gpu/arm/bifrost/csf/mali_kbase_csf_trace_buffer.c
@@ -88,7 +88,7 @@ struct firmware_trace_buffer {
 	} cpu_va;
 	u32 num_pages;
 	u32 trace_enable_init_mask[CSF_FIRMWARE_TRACE_ENABLE_INIT_MASK_MAX];
-	char name[]; /* this field must be last */
+	char name[1]; /* this field must be last */
 };
 
 /**
-- 
2.43.0

